### PR TITLE
fix(a11y): match the DOM order of the Modal's title and close button with their visual order

### PR DESCRIPTION
**Pull request by sis0k0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The visual order in the Modal/Wizard is the following:
1. Title
2. Close button

The DOM order is:
1. Close button
2. Title

Issue Number: #4083 

## What is the new behavior?

Now, the DOM order matches the visual order:
1. Title
2. Close button

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

##

ToDo:

- [x] Add tests
- [x] Verify that this PR doesn't introduce any breaking changes.

### CODE REVIEW

**Comment by mathisscott** [comment]
@sis0k0 and @bdryanovski 

I vote to `Handle this change only inside the clr-modal template and update the documentation on the best a11y pattern on the position of the elements.`

**Comment by mathisscott** [comment]
Are we going to get this merged in for the v4 release?

**Comment by sis0k0** [comment]
Sounds good to me! @bdryanovski, what do you think?

File: packages/angular/projects/clr-angular/src/modal/modal.spec.ts [hunk 1]
```diff
@@ -277,4 +277,14 @@ describe('Modal', () => {
     expect(messages[0].innerText).toBe('Beginning of Modal Content');
     expect(messages[1].innerText).toBe('End of Modal Content');
   }));
+
+  it('renders the title before the close button', fakeAsync(() => {
```
Comment (Jinnie) [inline packages/angular/projects/clr-angular/src/modal/modal.spec.ts:281]: Not sure it needs the fakeAsync.

**Comment by sis0k0** [comment]
@bdryanovski, I've updated the PR as we discussed.

**Review by bdryanovski — APPROVED** [review]

**Comment by mathisscott** [comment]
@sis0k0 
Are we going to merge this for the release tomorrow?

(Both `master` and `v4` branches...)

**Comment by sis0k0** [comment]
@mathisscott, I don't see any blockers for that. I'll make sure we have it in both branches today.

**Pull request merged**